cpp/ql/src/Likely Bugs/Format/WrongTypeFormatArguments.ql

```diff
@@ -25,7 +25,8 @@ private predicate formattingFunctionCallExpectedType(FormattingFunctionCall ffc,
       ffc.getTarget() = f and
       f.getFormatParameterIndex() = i and
       ffc.getArgument(i) = fl and
-      fl.getConversionType(pos) = expected
+      fl.getConversionType(pos) = expected and
+      count(fl.getConversionType(pos)) = 1
```

Comment:
I suspect this is only needed because the `where` clause in this query has its logic the wrong way around for when there are multiple `expected` types. I'd like to try fixing this, but I won't let it block this PR.

Reply:
I will look at this in a follow-up.

Reply:
To clarify, I'm thinking that

```
not trivialConversion(expected.getUnspecifiedType(), actual.getUnspecifiedType())
```

should be replaced by

```
not exists(Type anyExpected |
  formatArgType(ffc, n, anyExpected, arg, actual) and
  trivialConversion(anyExpected.getUnspecifiedType(), actual.getUnspecifiedType())
)
```

If it's a UI issue to have multiple alert messages for the same line, we could use `strictconcat` to concatenate the names of the expected types instead of having a `Type expected` in the `from` clause.